test/cpp/jit/test_gpu.cpp

```diff
@@ -8217,6 +8220,54 @@ TEST(NVFuserTest, FusionMagicSchedulerSoftmax_CUDA) {
       lparams);
 }
 
+TEST(NVFuserTest, TestMaskSoftmax_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  const int kReductionAxis = 3;
+  std::vector<int64_t> input_shape{256, 16, 128, 128};
+  TensorView* input = makeSymbolicTensor(input_shape.size());
+  TensorView* mask = makeSymbolicTensor(input_shape.size());
+  fusion.addInput(input);
+  fusion.addInput(mask);
+
+  auto out1 = add(input, mask);
+  auto output = softmax(out1, kReductionAxis);
+
+  fusion.addOutput(output);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor aten_input = at::randn(input_shape, options);
+  at::Tensor aten_mask = at::ones(input_shape, options);
+  aten_mask = aten_mask * -10000.0;
```

Comment:
Maybe add a comment why there is a `-10000` here instead of `std::numeric_limits<float>::lowest()`.

Reply:
Added appropriate comments.  Normally, we would use -Infinity because e^-Inf == 0.  However, Bert sometimes adds whole rows of padding that can result in a summation of all zeros resulting in a NaN.  Hence, a small finite number is used.